terminator/lib/queue-handlers.js

```diff
@@ -0,0 +1,196 @@
+    `,
+    )
+      .bind(hash, dataSetId)
+      .run()
+
+    // Start transaction monitor workflow
+    await env.TRANSACTION_MONITOR_WORKFLOW.create({
+      id: `transaction-monitor-${hash}-${Date.now()}`,
+      params: {
+        transactionHash: hash,
+      },
+    })
+
+    console.log(`Started transaction monitor workflow for transaction: ${hash}`)
+  } catch (error) {
+    console.error(
+      `Failed to process terminate CDN service for dataSetId: ${dataSetId}`,
+      error,
+    )
+    throw error
+  }
+}
+
+/**
+ * Handles transaction cancellation queue messages
+ *
+ * @param {TransactionCancelMessage} message
+ * @param {Env} env
+ */
+export async function handleTransactionCancelQueueMessage(
+  message,
+  env,
+  { getChainClient = defaultGetChainClient } = {},
+) {
+  const { transactionHash } = message
+
+  console.log(
+    `Processing transaction cancellation for hash: ${transactionHash}`,
+  )
+
+  try {
+    const { publicClient, walletClient } = getChainClient(env)
+
+    // First check if the original transaction is still pending
+    try {
+      const receipt = await publicClient.getTransactionReceipt({
+        hash: transactionHash,
+      })
+
+      if (receipt && receipt.blockNumber && receipt.blockNumber > 0n) {
+        console.log(
+          `Transaction ${transactionHash} is no longer pending, cancellation not needed`,
+        )
+        return
+      }
+    } catch (error) {
+      // Transaction not found or still pending, continue with cancellation
+      console.log(
+        `Transaction ${transactionHash} is still pending, proceeding with cancellation`,
+      )
+    }
+
+    // Get the original transaction to determine gas price
+    const originalTx = await publicClient.getTransaction({
+      hash: transactionHash,
+    })
+
+    console.log(
+      `Retrieved original transaction ${transactionHash} for cancellation`,
+    )
+
+    // Calculate higher gas price (150% of original)
+    const newGasPrice = originalTx.gasPrice
+      ? (originalTx.gasPrice * 3n) / 2n
+      : undefined
+    const newMaxFeePerGas = originalTx.maxFeePerGas
+      ? (originalTx.maxFeePerGas * 3n) / 2n
+      : undefined
+    const newMaxPriorityFeePerGas = originalTx.maxPriorityFeePerGas
+      ? (originalTx.maxPriorityFeePerGas * 3n) / 2n
+      : undefined
+
+    // Send empty transaction to same address with same nonce but higher gas to cancel
+    const cancelHash = await walletClient.sendTransaction({
+      to: originalTx.to,
+      value: 0n,
+      nonce: originalTx.nonce,
+      gasPrice: newGasPrice,
+      maxFeePerGas: newMaxFeePerGas,
+      maxPriorityFeePerGas: newMaxPriorityFeePerGas,
+    })
+
+    console.log(
+      `Sent cancellation transaction ${cancelHash} for original transaction ${transactionHash}`,
+    )
+
+    await env.DB.prepare(
+      `UPDATE data_sets SET terminate_service_tx_hash = ? WHERE terminate_service_tx_hash = ?`,
+    )
+      .bind(cancelHash, transactionHash)
+      .run()
```

Comment:
```suggestion
    await env.DB.prepare(
      `UPDATE data_sets SET terminate_service_tx_hash = ? WHERE terminate_service_tx_hash = ?`,
    )
      .bind(cancelHash, transactionHash)
      .run()
```

This is confusing, as now we're storing the hash of a cancel transaction - but nothing tells me as a viewer that this is a cancel transaction and not a terminate transaction

Reply:
> Maybe we shouldn't cancel TXs, but resend the same TX with higher gas allowance.

Yes, I think that's actually a better approach. I've updated the handlers to retry the transaction rather then canceling it.